samples/ClassroomLabs/Modules/Library/Tools/LabCreationLibrary.psm1

```diff
@@ -338,29 +338,55 @@ function Publish-Labs {
                 $ConfigObject
             )
 
-            $lacs = $ConfigObject | Select-Object -Property ResourceGroupName, LabAccountName -Unique
+            $lacs = $ConfigObject | Select-Object -Property ResourceGroupName, LabAccountName, SharedGalleryResourceGroupName, SharedGalleryName -Unique
+
             Write-Host "Operating on the following Lab Accounts:"
             Write-Host $lacs
 
             $block = {
-                param($path, $ResourceGroupName, $LabAccountName)
+                param($path, $ResourceGroupName, $LabAccountName, $SharedGalleryResourceGroupName, $SharedGalleryName)
 
                 Set-StrictMode -Version Latest
                 $ErrorActionPreference = 'Stop'
 
                 $modulePath = Join-Path $path '..\Az.LabServices.psm1'
                 Import-Module $modulePath
 
-                if ((Get-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName) -eq $null ){
-                    New-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName | Out-Null
+                $labAccount = Get-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName
+
+                if ($labAccount -eq $null ){
+                    $labAccount = New-AzLabAccount -ResourceGroupName $ResourceGroupName -LabAccountName $LabAccountName | Out-Null
+                    Write-Host "$LabAccountName lab account created."
+                }
+                else {
+                    Write-Host "$LabAccountName lab account found - skipping create."
+                }
+
+                if ($SharedGalleryResourceGroupName -ne $null && $SharedGalleryName -ne $null){
+
+                    $gallery = $labAccount | Get-AzLabAccountSharedGallery
+                    if ($gallery -ne $null) {
+                        Write-Host "$LabAccountName lab account already has attached gallery $gallery."
+                    }
+                    else {
+                        $gallery = Get-AzGallery -ResourceGroupName $SharedGalleryResourceGroupName -Name $SharedGalleryName
+
+                        if ($gallery -ne $null) {
+                         Write-Host "$SharedGalleryName shared gallery found."
+                            New-AzLabAccountSharedGallery -LabAccount $labAccount -SharedGallery $gallery
```

Comment:
It might be a good idea to do this by Resource ID instead of by object...  The reason is because if the SIG is in a different subscription- we don't want to manually swap subscriptions back and forth to get the object if we don't have to (and I think the API just needs the SIG ID)

Reply:
Good catch - I've changed it to look by resource id...but, i am still having issues with it working to find a SIG successfully across subs.  I've tested it in my subs, but I have different tentants involved.  Also, tried it with Roger where we used our Microsoft subs, and still couldn't get it to work.  I'd like to talk to you more to see what I'm missing.